### PR TITLE
Hoisting exercise complete

### PR DESCRIPTION
**Pull request by gangadhar5613**

### CODE REVIEW

File: 2-higher-order-functions/eloquent/exercise.js [hunk 1]
```diff
@@ -2,9 +2,14 @@
...
 
 // Your code here.
+console.log(arrays.reduce((a,b)=>{return a.concat(b)}));
+
 // → [1, 2, 3, 4, 5, 6]
 
 // Challenge 2. Your own loop
+function loop( start, test, update, body ){
+	for(let value = start; test(value); value = update(value) ) body(value);
+}
```
Comment (nnnkit) [inline 2-higher-order-functions/eloquent/exercise.js]: Don't write code in one line! Indent properly.

File: 2-higher-order-functions/eloquent/exercise.js [hunk 2]
```diff
@@ -13,8 +18,14 @@ loop(3, n => n > 0, n => n - 1, console.log);
...
 
 // Challenge 3. Everything
+
 function every(array, test) {
   // Your code here.
+  for ( n = 0; n < array.length; n++){
+    if ( test(array[n]) == false  ) return false;
+  }
+  return true
+
```
Comment (nnnkit) [inline 2-higher-order-functions/eloquent/exercise.js]: What is n? Define `n`

File: practice/closure.md [hunk 3]
```diff
@@ -23,6 +37,9 @@ const final = name("Smith"); // final should be "Will Smith"
 ```js
 function isInBetween(a, b) {
   // your code goes here
+  return function(num){
+    return (a,b) ?  a >= num && num >= b : a <= num && b >= num;
+  }
```
Comment (nnnkit) [inline practice/closure.md]: Fin the logic error!

Comment (nnnkit) [reply practice/closure.md]: Write code that human understands.

File: practice/closure.md [hunk 4]
```diff
@@ -65,6 +89,14 @@ cricket(); // Your score of Cricket is 2
...
 function getCard(suit) {
   // your code goes here
+    let options = ["club", "spade", "heart", "dimond"]
+  if(options.includes(suit.toLowerCase())) {
+    let cards = [2,3,4,5,6,7,8,9,10,"J", "Q", "K", "A"];
+    return function(){
+      let randomCard = Math.floor(Math.random() * cards.length);
+      return `Card is: ${cards[random]} ${suit}`;
+    }
+  }
```
Comment (nnnkit) [inline practice/closure.md]: Change random to `randomCard`

File: practice/hoisting.md [hunk 5]
```diff
@@ -6,47 +6,50 @@
...
 // Output or Error Message
+//undefined;
 ```
 
+
 ```js
 console.log(animal);
 let animal = "monkey";
-// Output or Error Message
+// reference error;
```
Comment (nnnkit) [inline practice/hoisting.md]: Uncaught ReferenceError: Cannot access 'animal' before initialization

File: practice/hoisting.md [hunk 6]
```diff
@@ -6,47 +6,50 @@
...
+//  error as SayHello is  not defined
 ```
 
 ```js
 sayHello("Hey Everyone");
 let sayHello = msg => {
   alert(msg);
 };
+// error as not defined
 ```
```
Comment (nnnkit) [inline practice/hoisting.md]: DO it again!

**Review by nnnkit — CHANGES_REQUESTED** [review]
Make the required changes and push again and merge it

**Review by gangadhar5613 — COMMENTED** [review]
Accept the changes and completed changed and merged it to main branch

**Pull request closed without merging**